### PR TITLE
10 - Implement article sorting

### PR DESCRIPTION
**Pull request by Tiny-B**
This pr implements sorting of articles, it can sort based on date, votes or comment count in ascending or descending order, there is an apply button that will sort the articles with drop down selections to choose what to sort by

### CODE REVIEW

File: src/api.js [hunk 1]
```diff
@@ -4,8 +4,10 @@ const newsAPI = axios.create({
 	baseURL: 'https://nc-news-xd69.onrender.com/api',
 });
 
-const fetchArticles = async (sortQueryString = 'created_at') => {
-	const results = await newsAPI.get(`/articles?sort_by=${sortQueryString}`);
+const fetchArticles = async (
+	defaultSortString = '?sort_by=created_at&order_by=desc'
+) => {
+	const results = await newsAPI.get(`/articles${defaultSortString}`);
```
Comment (callumu21) [inline src/api.js]: I would change this to make use of Axios params. If you don't pass in a sort_by or order_by, Axios will ignore them when composing the query string automatically.

File: src/components/page-components/Home.jsx [hunk 2]
```diff
@@ -1,11 +1,78 @@
...
+import { useSearchParams } from 'react-router-dom';
+import { useNavigate } from 'react-router-dom';
 
 const Home = () => {
+	const navigate = useNavigate();
+	const [articles, setArticles] = useState([]);
+	const [isLoading, setIsLoading] = useState(true);
+	const [searchParams, setSearchParams] = useSearchParams();
+
+	const sortArticles = ({ sortValue, orderValue }) => {
```
Comment (callumu21) [inline src/components/page-components/Home.jsx]: We shouldn't be sorting our articles using JS methods- this should be done by our back-end using the query string that we provide. You will need to refactor your useEffect to depend on the selected order and sort state you have in place, and then refetch the data whenever a new query is needed.

File: src/components/page-components/Home.jsx [hunk 3]
```diff
@@ -1,11 +1,78 @@
...
+	};
+
+	useEffect(() => {
+		navigate('.', { replace: true });
+	}, []);
+
+	useEffect(() => {
+		api.fetchArticles().then(res => {
+			//not sure why comment count is a string, might be a back end mistake ill fix later
+			res.map(article => {
```
Comment (callumu21) [inline src/components/page-components/Home.jsx]: The new array being returned by your map isn't being saved anywhere here, so you're just setting the response array as it was originally in state.

**Review by callumu21 — CHANGES_REQUESTED** [review]
Good start, Kief! At the moment, you're manually sorting your retrieved articles, but we should be having a new fetch request for the ordered/filtered data from our BE each time instead. I've left a comment below on where to address this, but please resubmit this PR for approval once you've changed how the sorting is working in your app 👍

**Review by B7Bprog — APPROVED** [review]
Hey Kief, the sorting works fine and this branch is good to merge. 👍 

Please make sure that after completing ticket 9, this sort will also work for the articles filtered by a topic.

**Pull request merged**